packages/mds-compliance/mds-compliance-engine.ts

```diff
@@ -304,7 +314,23 @@ function processPolicy(
           break
         }
         case 'speed':
-          compliance_acc.push({ rule, matches: [] })
+          {
+            const comp: Compliance & { matches: SpeedMatch[] | null } = processSpeedRule(
+              rule,
+              events,
+              geographies,
+              devices
+            )
+            compliance_acc.push(comp)
+            vehiclesToFilter.push(
+              ...(comp.matches
+                ? comp.matches.reduce((acc: MatchedVehicle[], match: SpeedMatch) => {
+                    acc.push(match.matched_vehicle)
+                    return acc
+                  }, [])
+                : [])
+            )
+          }
```

Comment:
All of the vehicles which are `matched` in here need to be counted in the `total_violations` field on line 349, and included in the `vehicles_in_violation` section. We don't need the entire compliance output for each rule, you should do the minified version in the same style as `count` rules (which basically just reports the measured number for each geography_id in a rule)